### PR TITLE
add support for reading secrets from a JSON file

### LINKED ISSUE #364: Allow JSON files as input for Secrets Stores and KVStores
Currently the `fastly.toml` [specification](https://www.fastly.com/documentation/reference/compute/fastly-toml/) allows you to pass in a JSON file for config stores (and earlier, also for dictionaries)

Something like that is extremely handy to be able to do, to e.g. share configuration across multiple fastly.tomls for environments, autogenerating these values from other scripts, not having to check them into CI etc.

Is that something you might be interested in adding back?

### PR DESCRIPTION
**Pull request by joeshaw**
This uses the same format as for the KV store, introduced in #365.

Fixes #364.

### CODE REVIEW

File: lib/src/config/secret_store.rs [hunk 1]
```diff
@@ -23,12 +23,66 @@ impl TryFrom<Table> for SecretStoreConfig {
                 });
             }
 
-            let items = items.as_array().ok_or_else(|| {
-                FastlyConfigError::InvalidSecretStoreDefinition {
-                    name: store_name.to_string(),
-                    err: SecretStoreConfigError::NotAnArray,
+            // Either the items here is from a top-level file with
```
Comment (cee-dub) [inline lib/src/config/secret_store.rs:26]:
```suggestion
            // Either the items here are from a top-level file with
```

File: lib/src/config/secret_store.rs [hunk 2]
```diff
@@ -23,12 +23,66 @@ impl TryFrom<Table> for SecretStoreConfig {
...
+            // "file" and "format" keys or it's an inline array.
+            // We try to parse either one of them to the same Vec<toml::Value>
+            // to allow them to run through the same validation path futher down
```
Comment (cee-dub) [inline lib/src/config/secret_store.rs:29]:
```suggestion
            // to allow them to run through the same validation path further down.
```

**Review by cee-dub — COMMENTED** [review]
Just a couple spelling corrections from me. I didn’t see any obvious issues with the Rust, but I’m still learning it, so I don’t feel qualified to give a ✅.

**Review by ulyssa — APPROVED** [review]
This looks good to me! :sparkles:

**Pull request merged**